src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/AvaloniaXamlIlLanguageParseIntrinsics.cs

```diff
@@ -198,6 +198,15 @@ public static bool TryConvert(AstTransformationContext context, IXamlAstValueNod
                 return ConvertDefinitionList(node, text, types, types.RowDefinitions, types.RowDefinition, "row definitions", out result);
             }
 
+            if (type.Equals(types.Classes))
+            {
+                var classes = text.Split(' ');
+                var classNodes = classes.Select(c => new XamlConstantNode(node, types.XamlIlTypes.String, c)).ToArray();
```

Comment:
Technically we have a specialized text node,

Reply:
Changed to use text node instead then.

Reply:
@kekekeks Actually it seems that changing to text node broke it, do I have to give it string type manually too?

Reply:
Yes, you have to. Text nodes are usually constructed during the initial AST parsing when type system isn't involved yet, so it gets assigned with `new XamlAstXmlTypeReference(lineInfo, XamlNamespaces.Xaml2006, "String");` as its type that gets resolved to a CLR type in the corresponding stage.

Reply:
I saw that and I hoped that it will be resolved to a CLR string. I guess this is too late in the pipeline. Seems to work now with explicit string type.